app/controllers/discovered_hosts_controller.rb

```diff
@@ -25,8 +26,19 @@ def index
 
   def show
     # filter graph time range
+    @general =    {:Highlights => [ "productname", "memorysize", "interface", "manufacturer", "architecture", "macaddress", "processorcount", "physicalprocessorcount", "discovery_boot"]}
+    @categories = {:Storage => ["blockdevice"],
+                   :Hardware => ["hardw", "manufacturer", "memo", "process"],
+                   :Software => ["bios", "os", "discovery"]}
```

Comment:
I wonder if you can make this configurable, my idea is that every category could be represented using *single* regexp. Then you can load these from Settings (create new four settings named `discovery_storage_CATEGORY`) and provide sane default values for them. Example for hardware: `^(hardw|manufacturer|memo|process)".

Please get rid of other extra fact name comarisons in the code. Everything should have been categorized by the four regexps. No other include/match/delete calls in the code.